src/GafferSceneUI/ShaderView.cpp

```diff
@@ -275,6 +283,12 @@ void ShaderView::sceneRegistrationChanged( const PrefixAndName &prefixAndName )
 	}
 }
 
+void ShaderView::rendererRegistrationChanged()
+{
+	m_registryUpdated = true;
+	plugDirtied( inPlug() );
```

Comment:
Do we need to call `plugDirtied()` here rather than calling `updateRenderer()/updateRendererState()` directly? `plugDirtied()` defers the update until an idle event because it can't change the graph, but that shouldn't apply here, and it also calls `updateScene()` which at first glance doesn't seem to be necessary either.

Reply:
Here, not using `plugDirtied()` prevent the shaderball render to update when the user re-enable it. Without it, it is necessary to edit a parameter plug to get the shaderball rendered. I couldn't make it refresh with the other functions.